engine/apps/alerts/models/alert_group.py

```diff
@@ -1978,14 +1978,52 @@ def notify_in_slack_enabled(self):
     def is_presented_in_slack(self):
         return self.slack_message and self.channel.organization.slack_team_identity
 
+    # TODO: once we've finished migrating slack_message._channel_id to slack_message.channel, we can enable this
+    # and refactor the related code
+    #
+    # @property
+    # def organization(self) -> Organization:
+    #     return self.channel.organization
+
+    # @property
+    # def slack_channel(self) -> typing.Optional["SlackChannel"]:
+    #     """
+    #     If the `AlertGroup` already has a `SlackMessage` associated with it, that means that we already
+    #     know the channel.
+
+    #     Otherwise, infer the Slack channel from the `ChannelFilter` associated with the `AlertGroup`.
+    #     """
+    #     if self.slack_message:
+    #         # NOTE: this isn't available yet because we haven't finished migrating slack_message._channel_id to
+    #         # slack_message.channel
+    #         return self.slack_message.channel
+    #     elif self.channel_filter and self.channel_filter.slack_channel:
+    #         return self.channel_filter.slack_channel
+    #     return self.organization.default_slack_channel
+
+    # @property
+    # def slack_channel_id(self) -> str | None:
+    #     """
+    #     If the `AlertGroup` already has a `SlackMessage` associated with it, that means that we already
+    #     know the channel.
+
+    #     Otherwise, infer the Slack channel ID from the `ChannelFilter` associated with the `AlertGroup`.
+    #     """
+    #     return self.slack_channel.slack_id if self.slack_channel else None
+    # END TODO:
```

Comment:
we can't really use this yet because we haven't yet migrated `SlackMessage._channel_id` (char field) to `SlackMessage.channel` (foreign key ref). I'm planning on doing this next week.

Once that is done, this becomes a lot cleaner because:
- `slack_message.channel` will be a `SlackChannel` (`_channel_id` hasn't been migrated to `channel`)
- `channel_filter.slack_channel` already was migrated
- `organization.default_slack_channel` already was migrated

In this case, we can just pass around `SlackChannel`s and not string references to channel IDs 🙂